src/common/src/main/java/fi/vrk/xrd4j/common/util/FileUtil.java

```diff
@@ -47,35 +48,38 @@ private FileUtil() {
     }
 
     /**
-     * Reads the contents of the file denoted by the path name. If the file doesn't
+     * Reads the contents of the file denoted by the path name. Prefers reading file
+     * as a resource and secondly as a file in file system. If the file doesn't
      * exist, an empty string is returned.
      *
      * @param filePath
-     *            file path
+     *            resource or file path
      * @return contents of the file
      */
     public static String read(String filePath) {
         try {
-            File file = getFile(filePath);
-            byte[] encoded = Files.readAllBytes(Paths.get(file.getAbsolutePath()));
+            byte[] encoded = readFileAsBytes(filePath);
             return new String(encoded, Charset.forName("UTF-8"));
         } catch (IOException | IllegalArgumentException e) {
             logger.error("Could not read file {}.", filePath, e);
             return "";
         }
     }
 
-    private static File getFile(String filePath) {
-        File file = new File(FileUtil.class.getClassLoader().getResource(filePath).getFile());
-
-        if (!file.exists()) {
-            logger.warn("Resource file is not found!");
-            file = new File(filePath);
+    private static byte[] readFileAsBytes(String filePath) throws IOException {
+        try (InputStream stream = Thread.currentThread().getContextClassLoader().getResourceAsStream(filePath)) {
+            byte[] result = new byte[stream.available()];
```

Comment:
available() should not be used like this, see https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#available--